cartographer/mapping/internal/3d/pose_graph_3d.cc

```diff
@@ -522,27 +522,29 @@ void PoseGraph3D::WaitForAllComputations() {
   // a WhenDone() callback.
   {
     common::MutexLocker locker(&work_queue_mutex_);
-    locker.Await(
-        [this]() REQUIRES(work_queue_mutex_) { return work_queue_ == nullptr; });
+    while (!locker.AwaitWithTimeout(
+        [this]() REQUIRES(work_queue_mutex_) { return work_queue_ == nullptr; },
+        common::FromSeconds(1.))) {
+      std::ostringstream progress_info;
```

Comment:
I thought we could use the same progress report as below, i.e. percentage of finished nodes. No need to switch to a different style here.

Reply:
done. ptal.